File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.dart
class Solution {
  int findMin(List<int> nums) {
    int left = 0;
    int right = nums.length - 1;

    while (left < right) {
      int mid = (right + left) ~/ 2;

      if (nums[mid] > nums[right]) {
        left = mid + 1;
      } else {
        right = mid;
      }
    }

    return nums[left];
  }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: list[int]) -> int:
        point1 = 0
        point2 = 1 
        for i in range(len(nums) - 1):
            if nums[point1] < nums[point2]:
                point1 += 1
                point2 += 1
            else:
                return nums[point2]
        return nums[0]